internal/metrics/datanode.go

```diff
@@ -0,0 +1,240 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"strconv"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/milvus-io/milvus/internal/util/typeutil"
+)
+
+const (
+	// TODO: use the common status label
+	DataNodeMetricLabelSuccess = "success"
+	DataNodeMetricLabelFail    = "fail"
+	DataNodeMetricLabelTotal   = "total"
+
+	DataNodeMsgTypeInsert = "insert"
+	DataNodeMsgTypeDelete = "delete"
+)
+
+// TODO: move to metrics.go
+const nodeIDLabelName = "node_id"
+
+// TODO: move to metrics.go
+func GetCollectionLabel(collectionID typeutil.UniqueID) string {
+	return "collection_" + strconv.FormatInt(collectionID, 10)
+}
+
+// TODO: move to metrics.go
+func NodeID2Label(nodeID typeutil.UniqueID) string {
+	return strconv.FormatInt(nodeID, 10)
+}
+
+// dataNodeDurationBuckets involves durations in milliseconds
+var dataNodeDurationBuckets = []float64{0, 10, 100, 1000, 10000, 100000, 1000000, 10000000}
+
+var (
+	DataNodeNumFlowGraphs = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.DataNodeRole,
+			Name:      "num_flow_graphs",
+			Help:      "Number of flow graphs in DataNode.",
+		}, []string{
+			"num_flow_graphs",
+			nodeIDLabelName,
+		})
+
+	DataNodeConsumeMsgRowsCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.DataNodeRole,
+			Name:      "message_size_count",
+			Help:      "Messages size count consumed from msgStream in DataNode.",
```

Comment:
rows count consumed from msgStream in DataNode

Reply:
fixed